lib/router.js

```diff
@@ -421,8 +421,9 @@ Router.prototype._handlePaths = function(rootNode, spec, scope) {
 
     // Handle paths
     // Sequence the build process with `.each` to avoid race conditions
-    // while building the tree.
-    return P.each(Object.keys(paths), function(pathPattern) {
+    // while building the tree. Also sort(), so that fixed path segments sort
+    // before patterns (by virtue of `[a-zA-Z/] < '{'`).
+    return P.each(Object.keys(paths).sort(), function(pathPattern) {
```

Comment:
Honestly, this sorting seems to bring more problems than it tries to solve. Manually placing /robots.txt before /{blah}/foo seems like a saner approach for the time being, IMHO.

The correct way would probably be to assign importance to each path and sort based on that. If importance is not given or is equal, sort based on the number of path segments (fewer first), and then based on whether they have params/wildcards (those without first).